Makefile

```diff
@@ -0,0 +1,60 @@
+# SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+.PHONY: xrefcheck test lint clean bats all
+
+# Build target from the common utility Makefile
+MAKEU = $(MAKE) -C make/
+
+MAKE_PACKAGE = $(MAKEU) PACKAGE=xrefcheck
+
+xrefcheck:
+	$(MAKE_PACKAGE) dev
+
+# Runs all tests.
+test:
+	make test-unit
+	make test-links
+	make bats
+
+test-dumb-term:
+	$(MAKE_PACKAGE) test-dumb-term
+
+test-hide-successes:
+	$(MAKE_PACKAGE) test-hide-successes
+
+clean:
+	$(MAKE_PACKAGE) clean
+
+lint:
+	hlint .
+
+####################################
+# Individual test suites
+
+test-unit:
+	$(MAKEU) test PACKAGE="xrefcheck:test:xrefcheck-tests"
+
+test-links:
+	vsftpd \
+    -orun_as_launching_user=yes \
+    -olisten_port=2221 \
+    -olisten=yes \
+    -oftp_username=$(shell whoami) \
+    -oanon_root=./links-tests/ftp_root \
+    -opasv_min_port=2222 \
+    -ohide_file='{.*}' \
+    -odeny_file='{.*}' \
+    -oseccomp_sandbox=no \
+    -olog_ftp_protocol=yes \
+    -oxferlog_enable=yes \
+    -ovsftpd_log_file=/tmp/vsftpd.log &
+
+	$(MAKEU) test PACKAGE="xrefcheck:test:links-tests" \
+		TEST_ARGUMENTS="--ftp-host ftp://127.0.0.1:2221"
+
+bats:
```

Comment:
We should be able to pass in a test filter, like we can in coffer (e.g. `make bats BATSFILTER="test name"`)